packages/accounts-controller/src/AccountsController.ts

```diff
@@ -791,6 +791,7 @@ export class AccountsController extends BaseController<
         added: [] as {
           address: string;
           type: string;
+          entropySource: string;
```

Comment:
WDYT about changing the logic a bit, cause it seems that having `entropySource` for EVERY added accounts is a bit wrong to me.

Maybe we could do this instead:
```suggestion
          options: InternalAccount['options'];
```

Then we populate this options when we're calling `patch.added.push`, this way we can just use later on and spread it with the existing `options`?

Reply:
I still think the logic about re-creating `InternalAccount`s from normal accounts is not the best right now and we should only have 1 place where we "fixup"/"inject" those things.

But I'm ok with going with this options for now to unblock you.

Again, all this would be naturally fixed once we start using `KeyringAccount`s on every keyrings!

Reply:
I saw this `added` array as an intermediate step, since it doesn't reflect the output of the method.
The actual `options` for the `InternalAccount` are modified based on this `added` array later on, so it seemed like extra complications to try to match types here.